### PR TITLE
Refactor: skipif for Windows 2/n

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

using Windows skipif

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
would be easier if this goes after #6266

**Review by s-rog — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: tests/accelerators/test_ddp.py [hunk 1]
```diff
@@ -83,7 +82,7 @@ def test_cli_to_pass(tmpdir, args=None):
     return '1'
 
 
-@pytest.mark.skipif(platform.system() == "Windows", reason="Distributed training is not supported on Windows")
+@RunIf(skip_windows=True)
```
Comment (tchaton) [inline tests/accelerators/test_ddp.py:85]: I find this `RunIf` decorator confusing as you provide arguments to Skip.

**Review by tchaton — APPROVED** [review]
Overall lgtm, but I find it a little bit confusing.

File: tests/accelerators/test_ddp.py (hunk 1 above)
Comment (awaelchli) [reply tests/accelerators/test_ddp.py:85]: We could simply provide both RunIf and SkipIf. Sometimes it makes sense to use one or the other.

Comment (Borda) [reply tests/accelerators/test_ddp.py:85]: yes, but then is miss the beauty of having a one-line configuration...

**Pull request merged**